internal/jobservice/application.go

```diff
@@ -0,0 +1,90 @@
+package jobservice
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/G-Research/armada/internal/common/auth/authorization"
+	grpcCommon "github.com/G-Research/armada/internal/common/grpc"
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	"github.com/G-Research/armada/internal/jobservice/repository"
+
+	"github.com/G-Research/armada/internal/jobservice/server"
+
+	js "github.com/G-Research/armada/pkg/api/jobservice"
+)
+
+type App struct {
+	// Configuration for jobService
+	Config *configuration.JobServiceConfiguration
+}
+
+func New(config *configuration.JobServiceConfiguration) *App {
+	return &App{
+		Config: config,
+	}
+}
+
+func (a *App) StartUp(ctx context.Context) error {
+	config := a.Config
+
+	// Setup an errgroup that cancels on any job failing or there being no active jobs.
+	g, _ := errgroup.WithContext(ctx)
+
+	grpcServer := grpcCommon.CreateGrpcServer(config.Grpc.KeepaliveParams, config.Grpc.KeepaliveEnforcementPolicy, []authorization.AuthService{&authorization.AnonymousAuthService{}})
+
+	inMemoryMap := make(map[string]*repository.JobTable)
+	subscribedJobSets := make(map[string]*repository.SubscribeTable)
+	jobStatusMap := repository.NewJobStatus(inMemoryMap, subscribedJobSets)
+	inMemoryJobService := repository.NewInMemoryJobServiceRepository(jobStatusMap, config)
```

Comment:
If the idea is to create and use a JobServiceRepository interface, we need to ensure these are typed to the interface, and not a specific implementation. This issue sorta exists throughout, but this seemed the most sensible place to comment it.

Reply:
Personally I don't think we need an interface here.  This component is pretty small and focused so I think it;'s better to be explicit rather than trying to abstract things away

Reply:
I went ahead and removed it.